app/controllers/concerns/hal_actions.rb

```diff
@@ -59,11 +59,10 @@ def zoom_param
     }.call
   end
 
-
   def index_cache_path
     index_resources = self.try(:resources_base) || resources
     PagedCollection.new(
-      with_params(self.class.resources_params, index_resources),
+      decorate_query(index_resources),
```

Comment:
Yes. That is the part that creates a unique cache key based on the items in the collection, though I see I need to add a position integer to it so we handle order changes, but otherwise looks right.